config/v1/types_infrastructure.go

```diff
@@ -691,7 +691,44 @@ type BareMetalPlatformLoadBalancer struct {
 
 // BareMetalPlatformSpec holds the desired state of the BareMetal infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type BareMetalPlatformSpec struct{}
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.apiServerInternalIPs) || has(self.apiServerInternalIPs)", message="Value is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.ingressIPs) || has(self.ingressIPs)", message="Value is required once set"
+type BareMetalPlatformSpec struct {
+	// apiServerInternalIPs are the IP addresses to contact the Kubernetes API
+	// server that can be used by components inside the cluster, like kubelets
+	// using the infrastructure rather than Kubernetes networking. These are the
+	// IPs for a self-hosted load balancer in front of the API servers.
+	// In dual stack clusters this list contains two IP addresses.
+	// In single stack clusters a single IP address is expected.
+	// When omitted, values from the status.apiServerInternalIPs will be used.
+	// Once set, the list cannot be completely removed (but its second entry can).
+	//
+	// +kubebuilder:validation:Format=ip
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=set
+	// +optional
+	APIServerInternalIPs []string `json:"apiServerInternalIPs"`
+
+	// ingressIPs are the external IPs which route to the default ingress
+	// controller. The IPs are suitable targets of a wildcard DNS record used to
+	// resolve default route host names.
+	// In dual stack clusters this list contains two IP addresses.
+	// In single stack clusters a single IP address is expected.
+	// When omitted, values from the status.ingressIPs will be used.
+	// Once set, the list cannot be completely removed (but its second entry can).
+	//
+	// +kubebuilder:validation:Format=ip
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=set
+	// +optional
+	IngressIPs []string `json:"ingressIPs"`
+
+	// IP networks used to connect all the OpenShift cluster nodes. Each network
+	// is provided in the CIDR format, e.g. "192.0.2.0/24"
+	// +listType=set
+	// +optional
+	MachineNetworks []string `json:"machineNetworks"`
```

Comment:
Can we validate that this is a CIDR?

Reply:
Validating `is an address IPv4 or IPv6` via regex is some hardcore piece of code, like
```
'/^(?>(?>([a-f0-9]{1,4})(?>:(?1)){7}|(?!(?:.*[a-f0-9](?>:|$)){8,})((?1)(?>:(?1)){0,6})?::(?2)?)|(?>(?>(?1)(?>:(?1)){5}:|(?!(?:.*[a-f0-9]:){6,})(?3)?::(?>((?1)(?>:(?1)){0,4}):)?)?(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9]?[0-9])(?>\.(?4)){3}))$/iD'
```

I thought about it, then looked how other APIs validate it (hint: they don't really) and decided not to do it directly here

Reply:
The best existing out there seems to be `+kubebuilder:validation:Format=cidr` so I will go with that

Reply:
Can we test that format CIDR actually works?

Reply:
Seems like it does not work, the same way that `format=ip` does not (:

Reply:
After all used `[]CIDR` with a monster regex validation